src/Storages/StorageDistributed.cpp

```diff
@@ -925,7 +903,7 @@ void registerStorageDistributed(StorageFactory & factory)
         String remote_table = engine_args[2]->as<ASTLiteral &>().value.safeGet<String>();
 
         const auto & sharding_key = engine_args.size() >= 4 ? engine_args[3] : nullptr;
-        const auto & storage_policy = engine_args.size() >= 5 ? engine_args[4]->as<ASTLiteral &>().value.safeGet<String>() : "";
+        const auto & storage_policy = engine_args.size() >= 5 ? engine_args[4]->as<ASTLiteral &>().value.safeGet<String>() : "default";
```

Comment:
Actually this wasn't the "redundant code" (f5af64514f511f534f5), now it uses default storage policy even if the 5-th argument is not passed, while previous behavior creates separate volume with just `path` from `config.xml`